### PR TITLE
Handle unauthenticated requests by redirecting to login page. Display…

### PR DESCRIPTION
**Pull request by CalPinSW**
… use image when authenticated.

### CODE REVIEW

File: backend/src/app.py [hunk 1]
```diff
@@ -19,10 +21,22 @@
...
+    resp.delete_cookie("user_id")
+    return resp
+
+
 @app.route("/")
 def index():
     user_id = request.cookies.get("user_id")
     access_token = request.cookies.get("spotify_access_token")
+    if access_token is None:
+        raise UnauthorizedException
```
Comment (JackMead) [inline backend/src/app.py:39]: Nothing wrong with this, but it strikes me as some logic that:
* Is going to end up duplicated
* Is going to end up missed somewhere

Would this maybe be a better check to put directly on the BearerAuth class, which we trust all Spotify requests will use?

File: backend/src/dataclasses/playlist_info.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ class SimplifiedPlaylist(BaseModel):
     id: str
     name: str
     description: str
-    images: List[Image]
+    images: Optional[List[Image]] = None
```
Comment (JackMead) [inline backend/src/dataclasses/playlist_info.py:27]: Ah, you found it too!

File: frontend/src/api/index.ts [hunk 3]
```diff
@@ -23,6 +32,9 @@ export const getPlaylists = async (
     )}&offset=${encodeURIComponent(offset)}`,
     { credentials: "include" }
   );
+  if (response.status === 401) {
```
Comment (JackMead) [inline frontend/src/api/index.ts:35]: Ooh nice, I haven't tested this branch but this looks like it would improve the initial UX definitely!

**Review by JackMead — APPROVED** [review]
Some nice additions here 😄

File: backend/src/app.py (hunk 1 above)
Comment (CalPinSW) [reply backend/src/app.py:39]: Ah, that's a much cleaner way of doing it! Good shout!

**Pull request merged**